### PR TITLE
C#: introduce common.props, set LangVersion to 7.2

### PR DESCRIPTION
**Pull request by jtattermusch**
Prerequisite for https://github.com/grpc/grpc/pull/24530.

- get rid of Common.csproj.include and use Directory.Build.props and Directory.Build.targets files to import the properties and definitions automatically.

- set LangVersion to 7.2 in all our C# projects (because it allows new lang constructs and there is no reason not to).

### CODE REVIEW

**Comment by MoienTajik** [comment]
Why not set the `LangVersion` to the `Latest`? In this way, there's no need to update the LangVersion with every new C# version.
Also, currently, the latest version of C# is 9.

File: src/csharp/build/common.props [hunk 1]
```diff
@@ -19,11 +19,8 @@
...
 
-  <!-- Needed for the net45 build to work on Unix. See https://github.com/dotnet/designs/pull/33 -->
-  <ItemGroup>
-    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
-    </PackageReference>
-  </ItemGroup>
+  <PropertyGroup>
+    <LangVersion>7.2</LangVersion>
```
Comment (MoienTajik) [inline src/csharp/build/common.props:23]: I think setting this to `Latest` is a better option.

Comment (jtattermusch) [reply src/csharp/build/common.props:23]: I think 7.2 is enough for now and it allows the tests to be more deterministic as we have different version of dotnet SDK on different platforms on our CI. I don't want to our LangVersion to change if we decide to upgrade dotnet SDK while performing maintenance of our testing infrastructure.

Comment (MoienTajik) [reply src/csharp/build/common.props:23]: Hmm, So there's no plan to add support for some features which exist in newer versions of C# like `null ref types` at least for now?

Comment (jtattermusch) [reply src/csharp/build/common.props:23]: Needs for other language features can be evaluated separately. For now we want C# 7.2 and I explained before why setting langversion explicitly makes sense.

**Review by apolcyn — APPROVED** [review]

**Pull request merged**